Maths/ArmstrongNumber.ts

```diff
@@ -0,0 +1,28 @@
+/** 
+ * @function ArmstrongNumber
+ * @description Check if the provided number is an Armstrong number or not.
+ * @summary Armstrong numbers are numbers, the sum of whose digits each raised
+ * to the power of the number of digits is equal to the number itself.
+ * For example:
+ * 370 is an Armstrong number since 3^3 + 7^3 + 0^3 = 370
+ * (These numbers are also known as Narcissistic numbers, and Pluperfect numbers)
+ * @param {number} num The number you want to check for
+ * @return {boolean} Whether the input number is an Armstrong number
+ * @see [Wikipedia](https://en.wikipedia.org/wiki/Armstrong_number)
+ * @see [OEIS](https://oeis.org/A005188)
+ * @example ArmstrongNumber(370) = true
+ * @example ArmstrongNumber(10) = false
+ */
+export const ArmstrongNumber = (num: number): boolean => {
+  if (typeof num !== 'number' || num <= 0) return false;
+
+  let compNum = 0
+
+  const digitArr = num
```

Comment:
I don't like using this to iterate over the digits of a number. Instead, you should be using:
```ts
const digitCount = Math.floor(1 + Math.log10(number));
let raisedDigitSum: number = 0;
while (num != 0) {
    const digit = num % 10;
    raisedDigitSum += digit ^ digitCount;
    num -= digit;
    num /= 10;
}
return raisedDigitSum == num;
```

Reply:
Hmmm, point understood. I'll fix it and commit it tomorrow then.

Reply:
Fixed (couldn't do earlier due to exams)